src/topic/__tests__/TopicContainer.test.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { configure, shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import configureMockStore from 'redux-mock-store';
+import 'jest-styled-components';
+import TopicContainer from '../TopicContainer';
+import thunk from 'redux-thunk';
+import TopicModel from '../../shared/models/TopicModel';
+import TopicsList from '../components/TopicsList';
+
+const mockedActionType = 'TopicContainerAction';
+
+jest.mock('../../actions/topic.actions', () => ({
+    getTopics: () => ({ type: mockedActionType })
+}));
+
+configure({ adapter: new Adapter() });
+const mockStore = configureMockStore([thunk]);
+
+describe('<TopicContainer>', () => {
+    const initialState = {
+        auth: {
+            loggedUser: false
+        },
+        topic: {
+            pending: true,
+            error: 'test error message',
+            topics: [TopicModel.fromBackendData({})]
+        }
+    };
+    let wrapper;
+    let store = mockStore(initialState);
+    
+    beforeEach(() => {
+        store.clearActions();
+        /**
+         * help would be appreciated. we have to (?) dive, otherwise we will not have the real actions binded to
+         * Component.
+         * 1) Passing it as props is a fake test (i.e we don't know if the real one is called
+         * 2) while using a dive, we have actions called immediately which may not be what we want
+         *
+         * I picked 2) as using 1) would only tell us that prop is called but we would lose the assurance that
+         * Container is correctly 'connected' (i.e. has correctly mapped dispatchToProps
+         *
+         */
+        wrapper = shallow(<TopicContainer store={store} />).dive();
+    });
+    
+    it('should match snapshot with error', () => {
+        expect(wrapper).toMatchSnapshot();
+    });
+    
+    it('should dispatch GET_TOPICS when mounted', () => {
+        expect(store.getActions()[0].type).toBe(mockedActionType);
+    });
+    
+    it('should have topics array connected correctly from redux', () => {
+        expect(wrapper.instance().props.topics).toBe(initialState.topic.topics);
```

Comment:
this still confuses me, `.props().topics` won't work here?

Reply:
The full example is provided there http://airbnb.io/enzyme/docs/api/ShallowWrapper/props.html
I think it is worth reading but to make it short: `wrapper.props()` returns only those props that are directly mentioned within the construction of `shallow(<Component props....)` - while we have some props that are `connect`ed by the Redux! thus:
```
const mapStateToProps = ({ topic }) => ({
    topics: topic.topics,
    pending: topic.pending,
    error: topic.error
});
```
no prop from those will be visible in `wrapper.props()`!

so in fact, we actually *could* pass all props instead of the store there... but it would be a test that doesnt actually test if our component builds correctly in connection to the Redux

Reply:
thanks for explanation!